app/shared/components/Contract/Interface/Form/Action.js

```diff
@@ -96,12 +106,47 @@ class ContractInterfaceFormAction extends Component<Props> {
       t,
       transaction
     } = this.props;
+    const {
+      currentArrayValues
+    } = this.state;
     const signing = !!(system.TRANSACTION_BUILD === 'PENDING');
     const fields = contract.getFields(contractAction);
-
     const formFields = [];
-    fields.forEach((field) => {
+    const fieldsMeta = fields.map((field) => {
+      if (field.type.substr(field.type.length - 2) === '[]') {
```

Comment:
@aaroncox do these changes also cover objects and look for "{" or "}"? I'm hearing from our dev that (potentially) the fix isn't fully complete as it doesn't look for "{" with "}" and "[" with "]".

From our dev:

> the greymass people did implement the support for arrays in the action data but they did it wrong
> well not really wrong but it won't work for all structs
> they check if the struct type contains '[]'
> but it's way better to check if the actual input contains [ and ] or { and}
> since structs can have custum names
> for example it won't work with our updateconfig
> because the type doesn't have [] in it
> it will work in our parser :smiley:

Reply:
Yeah it doesn't look like it fixes what you're looking for it to do. Our use case was the `names[]` type for things like the `voteproducer` action. 

Do you have an example of a live contract using the `{}` variant you're talking about, and what the final transaction output would look like? Even just a link to an actual `updateconfig` transaction might shed some light on it.

Reply:
Hey, @lukestokes asked me to reply here. Instead of checking if the type contains [] it's better to check if the actual user input contains [ and ] or { and }. I'm doing it like so...
![image](https://user-images.githubusercontent.com/44613132/53251457-f98c8c00-36bc-11e9-9b08-0f32931ca634.png)
the reason for this is that the type name of the struct can have a random name.

Reply:
@piecesnbits thanks for dropping by and helping expand on this!

So here's where I think this is different - currently we don't have a value, the value is empty as we're not processing a transaction that exists, but instead creating a new transaction that has never existed before. With the lack of a value the only the information we have to use is contained within the ABI structure returned from the chain itself.

Currently the code detects if the field is an array, and then creates a special UI element to help populate a JSON array (instead of users having to write in the JSON itself). That's what we were fixing in this PR.

From the sounds of it - what @lukestokes was asking about is a totally different issue. 

As I'm starting to understand this, what we need to add is additional code that detects if a user enters either a JSON array or object (like your sample above), and then parse that as such before creating the transaction. Is that right? If so we'll make sure to add that in our 0.7.3 release.

Reply:
Alright I think I still didn't understand what was happening when I responded above, but I'm starting to understand the issue a bit more. 

One thing that still would help is a link to a tx that contains the final result of one of these transactions. I'm still looking for a way to reproduce/test before I deploy the code live - and being about to simulate a tx for`thisaction` on the `eosdactokens` account  would help.

Reply:
I have https://github.com/greymass/eos-voter/pull/550/commits/3553e5304c7aa0aae3cfe13db3c47806e03d02d8 ready to release in 0.7.3 - which I think solves the issue we're talking about here.

Reply:
Excellent! :)

Reply:
I have found a way to validate user input in a more advanced way. I discovered this accidently. it is not perfect but it can handle validation of input fields where the struct is an object. it even detects if props are missing. The trick is to serialize the actiondata with eosjs@beta. if wrong input is supplied the serialization will error with a meaningful error code!

Reply:
![image](https://user-images.githubusercontent.com/44613132/53496906-4c8e8680-3aa3-11e9-887b-196470c86299.png)

Reply:
Awesome! We haven't moved to the newest eosjs yet - but I'll have to keep this method in mind once we make the change.